wayback-core/src/main/java/org/archive/wayback/util/webapp/RequestMapper.java

```diff
@@ -282,4 +295,168 @@ else if (prefix.endsWith("/") && (requestUrl + "/").equals(prefix)) {
+			RequestHandler handler) {
+		Matcher m = null;
+		m = HOST_PORT_PATTERN.matcher(name);
+		if (m.matches()) {
+			int port = Integer.parseInt(m.group(2));
+			addRequestHandler(port, m.group(1), null, handler);
+			return true;
+		}
+		return false;
+	}
+
+	/*
+	 * matches: localhost.archive.org:8080:two
+	 * static.localhost.archive.org:8080:fish
+	 */
+	private boolean registerHostPortPath(String name,
+			RequestHandler handler) {
+		Matcher m = null;
+		m = HOST_PORT_PATH_PATTERN.matcher(name);
+		if (m.matches()) {
+			int port = Integer.parseInt(m.group(2));
+			addRequestHandler(port, m.group(1), m.group(3), handler);
+			return true;
+		}
+		return false;
+	}
+
+	/*
+	 * matches: http://localhost.archive.org:8080/two the port is optional, and
+	 * need not be part of the URI if not included, using the internalPort
+	 * setting
+	 */
+	private boolean registerURIPatternPath(String name, int port,
+			RequestHandler handler) {
+		Matcher m = null;
+		m = URI_PATTERN.matcher(name);
+
+		if (m.matches()) {
+			String host = m.group(2);
+			String portString = m.group(3);
+
+			if ((portString != null) && portString.startsWith(":")) {
+				port = Integer.parseInt(portString.substring(1));
+			}
+
+			String path = m.group(4);
+
+			addRequestHandler(port, null, path, handler);
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * Extract the RequestHandler objects beanName, parse it, and register the
+	 * RequestHandler with the RequestMapper according to the beanNames
+	 * semantics.
+	 * @param handler The RequestHandler to register
+	 */
+	public void registerHandler(RequestHandler handler) {
+
+		String name = null;
+		int internalPort = 8080;
+
+		if (handler instanceof AbstractRequestHandler) {
+			name = ((AbstractRequestHandler)handler).getAccessPointPath();
+			internalPort = ((AbstractRequestHandler)handler).getInternalPort();
+		}
+
+		if (name == null) {
+			name = handler.getBeanName();
+		}
+
+		if (name != null) {
+			if (name.equals(RequestMapper.GLOBAL_PRE_REQUEST_HANDLER)) {
+				LOGGER
+					.info("Registering Global-pre request handler:" + handler);
+				addGlobalPreRequestHandler(handler);
+
+			} else if (name.equals(RequestMapper.GLOBAL_POST_REQUEST_HANDLER)) {
+
+				LOGGER.info("Registering Global-post request handler:" +
+						handler);
+				addGlobalPostRequestHandler(handler);
+
+			} else {
+				try {
+
+					boolean registered = registerPort(name, handler) ||
+							registerPortPath(name, handler) ||
+							registerHostPort(name, handler) ||
+							registerHostPortPath(name, handler) ||
+							registerURIPatternPath(name, internalPort, handler);
+
+					if (!registered) {
+						LOGGER.severe("Unable to register (" + name + ")");
+					}
+				} catch (NumberFormatException e) {
+					LOGGER.severe("FAILED parseInt(" + name + ")");
+				}
+			}
+		} else {
+			LOGGER.info("Unable to register RequestHandler - null beanName");
+		}
+//			if(handler instanceof ShutdownListener) {
```

Comment:
Remove it. Why is it useful to keep commented code?

Reply:
Let me keep these lines for a while, so that I can quickly remember the change when it blows up something as to ShutdownListener processing.